internal/controller/controller.go

```diff
@@ -223,30 +224,42 @@ func (c *Controller) startInformers(stopCh <-chan struct{}) {
 }
 
 func (c *Controller) createMeshServices() error {
-	services, err := c.clients.GetServices(metav1.NamespaceAll)
+	// Because createMeshServices is called after startInformers,
+	// then we already have the cache built, so we can use it.
+	svcs := c.kubernetesFactory.Core().V1().Services().Lister()
```

Comment:
Could we create a listWrapper struct that contains a map of listers, that we initialize in the controller `Init()`?

That way we could create an interface for the wrapper, and pass the interface to providers etc, and not have to use the kubeClient wrapper, therefore ensuring that we use the cache moving forward.

Reply:
Again, we might want to have that as a separate issue that we track and test separately in a new PR to prevent creep.

Reply:
I'm not sure I want to do this in this PR, this means lots of work.